offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/persons/CustomerDbConnector.groovy

```diff
@@ -270,15 +289,60 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
     return affiliationIds[0]
   }
 
+  private void setCustomerActive(int customerId, boolean active) {
+    String query = "UPDATE person SET active = ? WHERE id = ?";
+    
+    Connection connection = connectionProvider.connect()
+
+    connection.withCloseable {
+      def statement = connection.prepareStatement(query)
+      statement.setBoolean(1, active)
+      statement.setInt(2, customerId)
+      statement.execute()
+    }
+  }
+  
   /**
    * @inheritDoc
    * @param customerId
    * @param updatedCustomer
    */
   @Override
   void updateCustomer(String customerId, Customer updatedCustomer) {
-    //TODO implement
-    throw new RuntimeException("Method not implemented.")
+
+        int oldCustomerId = Integer.parseInt(customerId)
+    try {
+      if (getCustomer(oldCustomerId)==null) {
+        throw new DatabaseQueryException("Customer is not in the database and can't be updated.")
+      }
+            
+      Connection connection = connectionProvider.connect()
+      connection.setAutoCommit(false)
+
+      connection.withCloseable {it ->
+        try {
+          int newCustomerId = createNewCustomer(it, updatedCustomer)
+          storeAffiliation(it, newCustomerId, updatedCustomer.affiliations)
+          connection.commit()
+          
+          // if our update is successful we set the old customer inactive
+          setCustomerActive(oldCustomerId, false)
+          
+        } catch (Exception e) {
+          log.error(e.message)
+          log.error(e.stackTrace.join("\n"))
+          connection.rollback()
+          connection.close()
+          throw new DatabaseQueryException("Could not update customer.")
+        }
+      }
+    } catch (DatabaseQueryException ignored) {
+      throw new DatabaseQueryException("The customer could not be updated: ${customer.toString()}")
+    } catch (Exception e) {
+      log.error(e)
+      log.error(e.stackTrace.join("\n"))
+      throw new DatabaseQueryException("The customer could not be updated: ${customer.toString()}")
+    }
```

Comment:
I think we can reduce the number of exceptions here. The message does repeat and has no further value (redundant exception catching) How about just catching the exception Exception 1 time at the end of the block?

Reply:
The `ignored` exception should not be thrown. That is why it was called `ignored`

Reply:
good point, will change

Reply:
it seems the ignored exception was not supposed to be printed, but thrown. see the "add customer" method. I removed the case here now

Reply:
logged*